llvm/lib/Target/NVPTX/MCTargetDesc/NVPTXInstPrinter.cpp

```diff
@@ -319,6 +319,9 @@ void NVPTXInstPrinter::printLdStCode(const MCInst *MI, int OpNum,
     case NVPTX::PTXLdStInstCode::V4:
       O << ".v4";
       return;
+    case NVPTX::PTXLdStInstCode::V8:
```

Comment:
Why is this value stored in the instruction and printed via a flag? It seems like each instruction is only capable of representing one type of vector as the number of output operands is fixed.

Reply:
Yes, you're right, we can simplify this by removing the VecType concept and having that information live in tablegen.